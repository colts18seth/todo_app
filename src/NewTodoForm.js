import React, { useState } from 'react';
// import './NewTodoForm.css';

function NewTodoForm({ addTask }) {
    const INITIAL_STATE = {
        todo: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };

    const handleSubmit = e => {
        e.preventDefault();
        addTask(formData.todo);
        setFormData(INITIAL_STATE);
    }

    return (
        <form className="NewTodoForm" onSubmit={handleSubmit}>
            <input
                id="todo"
                name="todo"
                type="text"
                placeholder="Add New Task"
                value={formData.todo}
                onChange={handleChange}
            />
            <button>Add</button>
        </form>
    );
}

export default NewTodoForm;