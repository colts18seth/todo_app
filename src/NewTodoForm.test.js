import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders without crashing', () => {
    render(<NewTodoForm />);
});

test('matches snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

test('renders all parts of form', () => {
    const { getByText, getByPlaceholderText } = render(<NewTodoForm />);
    const input = getByPlaceholderText("Add New Task")
    const addButton = getByText("Add");
    expect(input).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
});