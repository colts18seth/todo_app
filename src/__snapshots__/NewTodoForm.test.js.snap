// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <input
      id="todo"
      name="todo"
      placeholder="Add New Task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
